flake.nix

```
@@ -0,0 +1,42 @@
+{
+  description = "Warnet python development environment";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+      in
+      {
+        devShells.default = pkgs.mkShell {
```

Maybe enough to do this since there aren't multiple?
```suggestion
        devShell = pkgs.mkShell {
```

---

I was reading how the implicit version is still here just for backwards compatibility. I wouldn't mind switching to it here since like you said, only one, but figured I'd get in the habit of being explicit.

---

That's a comment regarding the .lock-file, right, not the shell?

---

Ah sorry, I meant implicit default devShell, without the "default" key.

For the lock file, I was thinking the same as you that it might help avoid slightly different local versions.